subctl/issue-183-support-upgrading-submariner.md

```diff
@@ -0,0 +1,63 @@
+# Adding command to subctl to upgrade Submariner
+
+Related Issue: [Add subctl upgrade command](https://github.com/submariner-io/enhancements/issues/183)
+
+## Summary
+
+Currently, there is no automated way to upgrade Submariner. The user needs to manually upgrade `subctl`, since the version of Submariner
+to be installed is tied with the version of `subctl` installed, and then re-run `deploy-broker` and `join` sub-commands. This proposal
+describes a way to automate this process.
+
+## Proposal
+
+The proposal is to automate the process by adding a new `upgrade` sub-command that will do the above steps in the sequence.
+
+## Design details
+
+* subctl command name: `upgrade`
+* Flags for the command:
+  * `to`: mentions the version Submariner should be upgraded to. If not specified, upgrade to the latest stable version available.
+* Context to run on:
+  * Check on all contexts and re-run `deploy-broker` and `join` on clusters where Broker and Submariner are installed respectively.
+
+Once, the command is run, `subctl` will first be upgraded to the specified version. Once this is done, `subctl upgrade` will run again
```

Comment:
So are you saying that if we are using say 0.15.x version of `subctl` and invoke upgrade, the `subctl` **CLI itself** will first be updated to `0.16.0` (assuming we have a release) and then upgrade the deployment?